src/NuGet.Core/NuGet.Packaging/Definitions/IPackageDownloader.cs

```diff
@@ -56,6 +56,14 @@ public interface IPackageDownloader : IDisposable
         Task<string> GetPackageHashAsync(string hashAlgorithm, CancellationToken cancellationToken);
 
         /// <summary>
+        /// Sets an exception handler for package downloads.
+        /// </summary>
+        /// <param name="handleExceptionAsync">An exception handler.</param>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="handleExceptionAsync" />
+        /// is <c>null</c>.</exception>
+        void SetExceptionHandler(Func<Exception, Task<bool>> handleExceptionAsync);
```

Comment:
Elaborate on exception handler protocol: what value it should return and what expected behavior would be.